contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -0,0 +1,674 @@
+    //     @param _customFee performance fee charged by admin. Used as an integer percentage (i.e., 10 represents 10%). Does not accept "decimal" fees like "0.5".
+    //     @param _dataProvider id for getting the data provider contract address 0x1 to be passed.
+    //     @param _maxPlayersCount max quantity of players allowed to join the game
+    //     @param _incentiveToken optional token address used to provide additional incentives to users. Accepts "0x0" adresses when no incentive token exists.
+    //  */
+    constructor(
+        IERC20 _inboundCurrency,
+        ICurvePool _pool,
+        int128 _inboundTokenIndexInt,
+        uint256 _inboundTokenIndexUint,
+        uint256 _poolType,
+        ICurveGauge _gauge,
+        uint256 _segmentCount,
+        uint256 _segmentLength,
+        uint256 _segmentPayment,
+        uint256 _earlyWithdrawalFee,
+        uint256 _customFee,
+        uint256 _maxPlayersCount,
+        IERC20 _curve,
+        IERC20 _matic,
+        IERC20 _incentiveToken
+    ) public {
+        require(
+            _customFee <= 20,
+            "_customFee must be less than or equal to 20%"
+        );
+        require(
+            _earlyWithdrawalFee <= 10,
+            "_earlyWithdrawalFee must be less than or equal to 10%"
+        );
+        require(
+            _earlyWithdrawalFee > 0,
+            "_earlyWithdrawalFee must be greater than zero"
+        );
+        require(
+            _maxPlayersCount > 0,
+            "_maxPlayersCount must be greater than zero"
+        );
+        require(
+            address(_inboundCurrency) != address(0),
+            "invalid _inboundCurrency address"
+        );
+        require(_segmentCount > 0, "_segmentCount must be greater than zero");
+        require(_segmentLength > 0, "_segmentLength must be greater than zero");
+        require(
+            _segmentPayment > 0,
+            "_segmentPayment must be greater than zero"
+        );
+        require(address(_pool) != address(0), "invalid _pool address");
+        require(address(_gauge) != address(0), "invalid _gauge address");
+        require(address(_curve) != address(0), "invalid _curve address");
+        require(address(_matic) != address(0), "invalid _matic address");
+        // Initializes default variables
+        pool = _pool;
+        gauge = _gauge;
+        curve = _curve;
+        matic = _matic;
+        inboundTokenIndexInt = _inboundTokenIndexInt;
+        inboundTokenIndexUint = _inboundTokenIndexUint;
+        poolType = _poolType;
+        firstSegmentStart = block.timestamp; //gets current time
+        lastSegment = _segmentCount;
+        segmentLength = _segmentLength;
+        segmentPayment = _segmentPayment;
+        earlyWithdrawalFee = _earlyWithdrawalFee;
+        customFee = _customFee;
+        daiToken = _inboundCurrency;
+        maxPlayersCount = _maxPlayersCount;
+        incentiveToken = _incentiveToken;
+        if (_poolType == uint256(0)) {
```

Comment:
i'm interested to learn - why you cast `0` to `uint256`, but don't cast `1` below?

Reply:
oh, that's unnecessary will remove it